### ISSUE TITLE
integrated terminal not using default history file

### ISSUE THREAD

**Issue by oren-l** [microsoft/vscode#168767]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: Version: 1.74.0
Commit: 5235c6bb189b60b01b1f49062f4ffa42384f8c91
Date: 2022-12-05T16:43:37.594Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 21.6.0
Sandboxed: No
- OS Version: macOS Monterey 12.6 (21G115)

Steps to Reproduce:

1. open integrated terminal
2. run `echo $HISTFILE`

Expected: `/Users/<user>/.zsh_history` like in any other terminal app (iTerm, Termianl.app)

Actual: `/var/folders/bh/cjfwtj7d6kn102qd0q9w2m5h0000gp/T/<user>-code-zsh/.zsh_history`

This happened right after updating to latest version of vscode

**Comment by oren-l** [comment]
Solved this by manually setting the zsh args in my `settings.json` to include `-i` (force interactive mode)
I guess it used to be the default and in the latest version it may have changed.
```json
"terminal.integrated.profiles.osx": {
    "zsh": {
      "path": "zsh",
      "args": [
        "-il"
      ]
    }
  }
```

**Comment by oren-l** [comment]
also I started getting perl locale warnings only in the integrated terminal:
```zsh
$ git log -1
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LC_ALL = (unset),
        LC_CTYPE = "UTF-8",
        LC_TERMINAL = "iTerm2",
        LANG = "en_IL.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
```

solved it easily by adding these lines in my `.zshrc` file:
```config
export LC_CTYPE=en_US.UTF-8
export LC_ALL=en_US.UTF-8
```

but it may help with finding the source of the issue in vscode

**Comment by john-pimq** [comment]
Thanks for your solution @oren-l.
The warning message jump out when I type `yarn` and press tab key. Moreover, the warning only shows in vscode intergrated terminal.
I fixed the problem by just adding this two line in the bottom of my `~/.zshrc` file.
```zsh
export LC_CTYPE=en_US.UTF-8
export LC_ALL=en_US.UTF-8
```
Also, I found [this link from stackoverflow](https://stackoverflow.com/questions/2499794/how-to-fix-a-locale-setting-warning-from-perl) may be useful for other OS system user.
